packages/beacon-node/src/api/impl/lightclient/index.ts

```diff
@@ -12,42 +12,55 @@ export function getLightclientApi({
 }: Pick<ApiModules, "chain" | "config">): ApplicationMethods<routes.lightclient.Endpoints> {
   return {
     async getLightClientUpdatesByRange({startPeriod, count}) {
+      const lightClientServer = chain.lightClientServer;
+      assertLightClientServer(lightClientServer);
+
       const maxAllowedCount = Math.min(MAX_REQUEST_LIGHT_CLIENT_UPDATES, count);
       const periods = Array.from({length: maxAllowedCount}, (_ignored, i) => i + startPeriod);
-      const updates = await Promise.all(periods.map((period) => chain.lightClientServer.getUpdate(period)));
+      const updates = await Promise.all(periods.map((period) => lightClientServer.getUpdate(period)));
       return {
         data: updates,
         meta: {versions: updates.map((update) => config.getForkName(update.attestedHeader.beacon.slot))},
       };
     },
 
     async getLightClientOptimisticUpdate() {
-      const update = chain.lightClientServer.getOptimisticUpdate();
+      const lightClientServer = chain.lightClientServer;
+      assertLightClientServer(lightClientServer);
```

Comment:
re-assignment here is not needed, type assertion works either way

```suggestion
      assertLightClientServer(chain.lightClientServer);
```

Reply:
Nested attributes type assertion does not work.

> 'chain.lightClientServer' is possibly 'undefined'

Reply:
it only does not work if you use it in a callback, but here it works, in the other function above it does not